couchdb/server/state.py

```diff
@@ -2,11 +2,37 @@
 #
 import logging
 
-__all__ = ('add_fun', 'add_lib')
+__all__ = ('add_fun', 'add_lib', 'reset')
 
 log = logging.getLogger(__name__)
 
 
+def reset(server, config=None):
+    """Resets query server state.
+
+    :command: reset
+
+    :param server: Query server instance.
+    :type server: :class:`~couchdb.server.BaseQueryServer`
+
+    :param config: Optional dict argument to set up query config.
+    :type config: dict
+
+    :return: True
+    :rtype: bool
+    """
+    log.debug('Reset server state')
+    del server.state['functions'][:]
+    del server.state['functions_src'][:]
+    server.state['query_config'].clear()
+    if config is not None:
+        log.debug('Set new query config:\n%s', config)
+        server.state['query_config'].update(config)
+    if server.version >= (1, 1, 0):
+        server.state['view_lib'] = ''
+    return True
```

Comment:
Yep. The main concern in my question is that "If `state` get reset again and again, the value of `reduce_limilt` will be volatile. Should this value be basically consist?" 
I guess `reduce_limit` do not change in most of the time, so let user set it via command line option (just a boolean flag) make more sense to me.

FYI:
I am looking this code block in `BaseQueryServer`
```python
    def is_reduce_limited(self):
        """Checks if output of reduce function is limited."""
        return self.state['query_config'].get('reduce_limit', False)
```